bump bindgen

When we try to update the version of bindgen to a recent version we get the following error:

```
 error[E0308]: mismatched types
   --> crates/lisp/src/list.rs:273:18
    |
272 |     pub fn cdr(self) -> LispObject {
    |                         ---------- expected `LispObject` because of return type
273 |         unsafe { (*self._extract()).u.s.as_ref().u.cdr }
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `LispObject`, found struct `__BindgenUnionField`
    |
    = note: expected struct `LispObject`
               found struct `__BindgenUnionField<LispObject>`
```
#139

--- a/rust_src/crates/emacs/Cargo.toml
+++ b/rust_src/crates/emacs/Cargo.toml
@@ -13,7 +13,7 @@ lazy_static = "1.4.0"
 memoffset = "0.6.4"
 
 [build-dependencies]
-bindgen = "0.46"
+bindgen = "0.64"
 regex = "1.6"
 libc = "0.2"
 

--- a/rust_src/crates/emacs/build.rs.in
+++ b/rust_src/crates/emacs/build.rs.in
@@ -246,42 +246,42 @@ fn generate_bindings() {
         .no_copy("wr_output")
         // .blocklist_file("gtk/gtk.h")
         // we define these ourselves, for various reasons
-        .blacklist_item("Lisp_Object")
-        .blacklist_item("emacs_globals")
-        .blacklist_item("Q.*") // symbols like Qnil and so on
-        .blacklist_item("USE_LSB_TAG")
-        .blacklist_item("VALMASK")
-        .blacklist_item("PSEUDOVECTOR_FLAG")
-        .blacklist_item("Fmapc")
+        .blocklist_item("Lisp_Object")
+        .blocklist_item("emacs_globals")
+        .blocklist_item("Q.*") // symbols like Qnil and so on
+        .blocklist_item("USE_LSB_TAG")
+        .blocklist_item("VALMASK")
+        .blocklist_item("PSEUDOVECTOR_FLAG")
+        .blocklist_item("Fmapc")
         // these two are found by bindgen on mac, but not linux
-        .blacklist_item("EMACS_INT_MAX")
-        .blacklist_item("VAL_MAX")
+        .blocklist_item("EMACS_INT_MAX")
+        .blocklist_item("VAL_MAX")
         // this is wallpaper for a bug in bindgen, we don't lose much by it
         // https://github.com/servo/rust-bindgen/issues/687
-        .blacklist_item("BOOL_VECTOR_BITS_PER_CHAR")
+        .blocklist_item("BOOL_VECTOR_BITS_PER_CHAR")
         // this is wallpaper for a function argument that shadows a static of the same name
         // https://github.com/servo/rust-bindgen/issues/804
-        .blacklist_item("face_change")
+        .blocklist_item("face_change")
         // these never return, and bindgen doesn't yet detect that, so we will do them manually
-        .blacklist_item("error")
-        .blacklist_item("circular_list")
-        .blacklist_item("wrong_type_argument")
-        .blacklist_item("nsberror")
-        .blacklist_item("emacs_abort")
-        .blacklist_item("Fsignal")
-        .blacklist_item("memory_full")
-        .blacklist_item("wrong_choice")
-        .blacklist_item("wrong_range")
+        .blocklist_item("error")
+        .blocklist_item("circular_list")
+        .blocklist_item("wrong_type_argument")
+        .blocklist_item("nsberror")
+        .blocklist_item("emacs_abort")
+        .blocklist_item("Fsignal")
+        .blocklist_item("memory_full")
+        .blocklist_item("wrong_choice")
+        .blocklist_item("wrong_range")
         // these are defined in data.rs
-        // .blacklist_item("Lisp_Fwd")
-        // .blacklist_item("Lisp_.*fwd")
+        // .blocklist_item("Lisp_Fwd")
+        // .blocklist_item("Lisp_.*fwd")
         // these are defined in remacs_lib
-        .blacklist_item("timespec")
-        .blacklist_item("fd_set")
-        .blacklist_item("pselect")
-        .blacklist_item("sigset_t")
-        .blacklist_item("timex")
-        .blacklist_item("clock_adjtime")
+        .blocklist_item("timespec")
+        .blocklist_item("fd_set")
+        .blocklist_item("pselect")
+        .blocklist_item("sigset_t")
+        .blocklist_item("timex")
+        .blocklist_item("clock_adjtime")
         // by default we want C enums to be converted into a Rust module with constants in it
         .default_enum_style(bindgen::EnumVariation::ModuleConsts)
         // enums with only one variant are better as simple constants

--- a/rust_src/crates/emacs/src/list.rs
+++ b/rust_src/crates/emacs/src/list.rs
@@ -272,7 +272,7 @@ impl LispCons {
 
     /// Return the cdr (second cell).
     pub fn cdr(self) -> LispObject {
-        unsafe { (*self._extract()).u.s.as_ref().u.cdr }
+        unsafe { *(*self._extract()).u.s.as_ref().u.cdr.as_ref() }
     }
 
     /// Set the car of the cons cell.
@@ -285,7 +285,7 @@ impl LispCons {
     /// Set the cdr of the cons cell.
     pub fn set_cdr(self, n: impl Into<LispObject>) {
         unsafe {
-            (*self._extract()).u.s.as_mut().u.cdr = n.into();
+            *(*self._extract()).u.s.as_mut().u.cdr.as_mut() = n.into();
         }
     }
 

--- a/rust_src/crates/emacs/src/symbol.rs
+++ b/rust_src/crates/emacs/src/symbol.rs
@@ -40,7 +40,7 @@ impl LispSymbolRef {
 
     pub unsafe fn get_value(self) -> LispObject {
         let s = self.u.s.as_ref();
-        s.val.value
+        *s.val.value.as_ref()
     }
 
     pub const fn iter(self) -> LispSymbolIter {

--- a/rust_src/crates/lisp_macros/lib.rs
+++ b/rust_src/crates/lisp_macros/lib.rs
@@ -142,7 +142,7 @@ pub fn lisp_fn(attr_ts: TokenStream, fn_ts: TokenStream) -> TokenStream {
             subr_ref.min_args = #min_args;
             subr_ref.max_args = #max_args;
             subr_ref.symbol_name = (#symbol_name).as_ptr() as *const libc::c_char;
-            subr_ref.intspec.string = #intspec;
+            *subr_ref.intspec.string.as_mut() = #intspec;
             subr_ref.doc = 0;
 
                     std::ptr::copy_nonoverlapping(&subr, #srname.as_mut_ptr(), 1);

--- a/rust_src/crates/webrender/src/event.rs
+++ b/rust_src/crates/webrender/src/event.rs
@@ -1,3 +1,4 @@
+use crate::input::InputEvent;
 use emacs::{
     bindings::{event_kind, input_event, scroll_bar_part},
     globals::{Qnil, Qt},
@@ -5,8 +6,9 @@ use emacs::{
 };
 
 pub fn create_emacs_event(kind: event_kind::Type, top_frame: LispObject) -> input_event {
-    input_event {
-        _bitfield_1: input_event::new_bitfield_1(kind, scroll_bar_part::scroll_bar_nowhere),
+    InputEvent {
+        kind,
+        part: scroll_bar_part::scroll_bar_nowhere,
         code: 0,
         modifiers: 0,
         x: 0.into(),
@@ -16,4 +18,5 @@ pub fn create_emacs_event(kind: event_kind::Type, top_frame: LispObject) -> inpu
         arg: Qnil,
         device: Qt,
     }
+    .into()
 }

--- a/rust_src/crates/webrender/src/input.rs
+++ b/rust_src/crates/webrender/src/input.rs
@@ -38,11 +38,9 @@ impl InputProcessor {
             return None;
         }
 
-        let iev = input_event {
-            _bitfield_1: input_event::new_bitfield_1(
-                event_kind::ASCII_KEYSTROKE_EVENT,
-                scroll_bar_part::scroll_bar_nowhere,
-            ),
+        let iev: input_event = InputEvent {
+            kind: event_kind::ASCII_KEYSTROKE_EVENT,
+            part: scroll_bar_part::scroll_bar_nowhere,
             code: Self::remove_control(c) as u32,
             modifiers: Self::to_emacs_modifiers(self.modifiers),
             x: 0.into(),
@@ -51,7 +49,8 @@ impl InputProcessor {
             frame_or_window: top_frame,
             arg: Qnil,
             device: Qt,
-        };
+        }
+        .into();
 
         Some(iev)
     }
@@ -67,11 +66,9 @@ impl InputProcessor {
 
         self.suppress_chars = true;
 
-        let iev = input_event {
-            _bitfield_1: input_event::new_bitfield_1(
-                event_kind::NON_ASCII_KEYSTROKE_EVENT,
-                scroll_bar_part::scroll_bar_nowhere,
-            ),
+        let iev: input_event = InputEvent {
+            kind: event_kind::NON_ASCII_KEYSTROKE_EVENT,
+            part: scroll_bar_part::scroll_bar_nowhere,
             code: key_code as u32,
             modifiers: Self::to_emacs_modifiers(self.modifiers),
             x: 0.into(),
@@ -80,7 +77,8 @@ impl InputProcessor {
             frame_or_window: top_frame,
             arg: Qnil,
             device: Qt,
-        };
+        }
+        .into();
 
         Some(iev)
     }
@@ -107,11 +105,9 @@ impl InputProcessor {
             ElementState::Released => up_modifier,
         };
 
-        let iev = input_event {
-            _bitfield_1: input_event::new_bitfield_1(
-                event_kind::MOUSE_CLICK_EVENT,
-                scroll_bar_part::scroll_bar_nowhere,
-            ),
+        let iev: input_event = InputEvent {
+            kind: event_kind::MOUSE_CLICK_EVENT,
+            part: scroll_bar_part::scroll_bar_nowhere,
             code: c as u32,
             modifiers: Self::to_emacs_modifiers(self.modifiers) | s,
             x: (self.cursor_positon.x as i32).into(),
@@ -120,7 +116,8 @@ impl InputProcessor {
             frame_or_window: top_frame,
             arg: Qnil,
             device: Qt,
-        };
+        }
+        .into();
 
         Some(iev)
     }
@@ -188,9 +185,9 @@ impl InputProcessor {
         let (kind, is_upper, lines) = event_meta.unwrap();
 
         let s = if is_upper { up_modifier } else { down_modifier };
-
-        let iev = input_event {
-            _bitfield_1: input_event::new_bitfield_1(kind, scroll_bar_part::scroll_bar_nowhere),
+        let iev: input_event = InputEvent {
+            kind,
+            part: scroll_bar_part::scroll_bar_nowhere,
             code: 0,
             modifiers: Self::to_emacs_modifiers(self.modifiers) | s,
             x: (self.cursor_positon.x as i32).into(),
@@ -199,7 +196,8 @@ impl InputProcessor {
             frame_or_window: top_frame,
             arg: lines.into(),
             device: Qt,
-        };
+        }
+        .into();
 
         Some(iev)
     }
@@ -304,5 +302,35 @@ pub fn winit_keycode_emacs_key_name(keycode: VirtualKeyCode) -> *const libc::c_c
         VirtualKeyCode::F24 => kn!("f24"),
 
         _ => std::ptr::null(), // null pointer
+    }
+}
+
+pub struct InputEvent {
+    pub kind: event_kind::Type,
+    pub part: scroll_bar_part::Type,
+    pub code: ::libc::c_uint,
+    pub modifiers: ::libc::c_uint,
+    pub x: LispObject,
+    pub y: LispObject,
+    pub timestamp: emacs::bindings::Time,
+    pub frame_or_window: LispObject,
+    pub arg: LispObject,
+    pub device: LispObject,
+}
+
+impl From<InputEvent> for input_event {
+    fn from(val: InputEvent) -> Self {
+        let mut iev = input_event::default();
+        iev.set_kind(val.kind);
+        iev.set_part(val.part);
+        iev.code = val.code;
+        iev.modifiers = val.modifiers;
+        iev.x = val.x;
+        iev.y = val.y;
+        iev.timestamp = val.timestamp;
+        iev.frame_or_window = val.frame_or_window;
+        iev.arg = val.arg;
+        iev.device = val.device;
+        iev
     }
 }

